model/SearchEngine/Driver/Elasticsearch/ElasticSearchConfig.php

```diff
@@ -44,7 +44,25 @@ public function __construct(ServiceOptionsInterface $serviceOptions)
 
     public function getHosts(): array
     {
-        return $this->serviceOptions->get(self::class, self::OPTION_HOSTS);
+        $hosts = [];
+        foreach ($this->serviceOptions->get(self::class, self::OPTION_HOSTS) as $host) {
+            if (is_array($host) && isset($host['host'])){
+                $hosts[] = ($host['scheme'] ?? 'http') . '://' . $host['host'] . ':' . ($host['port'] ?? 9200);
+            } else {
+                $hosts[] = $host;
+            }
+        }
+        return $hosts;
+    }
+
+    public function getUsername(): ?string
```

Comment:
Can I kindly ask to cover the new methods in the test `ElasticSearchConfigTest` ?

Thanks! :)

Reply:
Done